### PR TITLE
feat: add Warning Modal triggers/connections/variables

### PR DESCRIPTION
**Pull request by J1za**
## Linear Ticket
https://linear.app/autokitteh/issue/UI-810/when-there-is-active-deployment-and-edit-asset-display-warning
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

**Comment by RonenMars** [comment]
@J1za - It should also pop up when we create a connection/trigger/variable.

When you do it, please take care of the E2E tests, which will probably break.

**Review by RonenMars — COMMENTED** [review]
@J1za - It should also pop up when we create a connection/trigger/variable.

When you do it, please take care of the E2E tests, which will probably break.

File: src/components/organisms/connections/integrations/asana/edit.tsx [hunk 1]
```diff
@@ -3,25 +3,39 @@ import React, { useState } from "react";
...
-
+	const { deployments } = useCacheStore();
+	const { openModal } = useModalStore();
 	const { errors, handleSubmit, isLoading, onSubmitEdit, register, setValue } = useConnectionForm(
 		asanaIntegrationSchema,
 		"edit"
 	);
 
+	const handleFormSubmit = () => {
+		if (deployments?.length && deployments[0].state === DeploymentStateVariant.active) {
```
Comment (RonenMars) [inline src/components/organisms/connections/integrations/asana/edit.tsx:29]: Don't check it each time in each component, but do something like this:
```
const initialState: Pick<
    CacheStore,
    "loading" | "deployments" | "triggers" | "variables" | "events" | "envId" | "currentProjectId" | "hasActiveDeployments"
> = {
    loading: {
        deployments: false,
        triggers: false,
        variables: false,
        events: false,
    },
    deployments: undefined,
    variables: [],
    triggers: [],
    events: undefined,
    currentProjectId: undefined,
    envId: undefined,
    hasActiveDeployments: false,
};

const store: StateCreator<CacheStore> = (set, get) => ({
    ...initialState,

    get hasActiveDeployments() {
        const { deployments } = get();
        return Boolean(
            deployments?.length && 
            deployments.some(deployment => deployment.state === DeploymentStateVariant.active)
        );
    },

```

And then use `hasActiveDeployments` as a getter in each component.

**Comment by RonenMars** [comment]
Please add the warning modal on connection/trigger/variable delete as well.

File: src/locales/en/modals/translation.json [hunk 2]
```diff
@@ -72,5 +72,11 @@
 		"readme": "README",
 		"nameRequired": "Name is required",
 		"nameTaken": "Name is already taken"
+	},
+	"warningActiveDeployment": {
+		"title": "Warning",
+		"warning": "This might affect running deployment",
```
Comment (RonenMars) [inline src/locales/en/modals/translation.json:78]:
```suggestion
		"warning": "Changes might affect the currently running deployments",
```

Comment (J1za) [reply src/locales/en/modals/translation.json:78]: changed

File: src/components/organisms/connections/integrations/asana/edit.tsx (hunk 1 above)
Comment (J1za) [reply src/components/organisms/connections/integrations/asana/edit.tsx:29]: added and refactored

File: src/components/organisms/triggers/deleteModal.tsx [hunk 3]
```diff
@@ -37,8 +39,8 @@ export const DeleteTriggerModal = ({ id, isDeleting, onDelete }: DeleteModalProp
 			<div className="mx-6">
 				<h3 className="mb-5 text-xl font-bold">{t("title")}</h3>
 				<p>{t("content", { name: trigger?.name })}</p>
-
-				<p>{t("deleteWarning")}</p>
+				<p className="mt-1">{t("deleteWarning")}</p>
+				{hasActiveDeployments ? <p className="mt-1 font-normal">{tWarning("warning")}</p> : null}
```
Comment (RonenMars) [inline src/components/organisms/triggers/deleteModal.tsx:43]: ### 
```suggestion
				{hasActiveDeployments ? <p className="mt-1 font-normal">{tWarning("warning")}</p> : null}
```
```suggestion
				{hasActiveDeployments ? <p className="mt-1 font-normal">{tWarning("hasActiveDeploymentsWarning")}</p> : null}
```

File: src/locales/en/modals/translation.json [hunk 4]
```diff
@@ -72,5 +72,11 @@
 		"readme": "README",
 		"nameRequired": "Name is required",
 		"nameTaken": "Name is already taken"
+	},
+	"warningActiveDeployment": {
+		"title": "Warning",
+		"warning": "Changes might affect the currently running deployments",
```
Comment (RonenMars) [inline src/locales/en/modals/translation.json:78]: Like in the other modals, the warning content should be content.

```suggestion
		"content": "Changes might affect the currently running deployments",
```

Comment (J1za) [reply src/locales/en/modals/translation.json:78]: changed

**Review by RonenMars — APPROVED** [review]

**Pull request merged**